rageval/metrics/_answer_bleu.py

```diff
@@ -0,0 +1,162 @@
+from dataclasses import dataclass
+from typing import List, Tuple
+
+import datasets
+from datasets import Dataset
+import re
+
+from rageval.metrics import Metric, add_attribute
+
+
+_DESCRIPTION = """\
+BLEU (Bilingual Evaluation Understudy) is an algorithm for evaluating the quality of text which has been machine-translated from one natural language to another.
+Scores are calculated for individual translated segments—generally sentences—by comparing them with a set of good quality reference translations.
+Those scores are then averaged over the whole corpus to reach an estimate of the translation's overall quality.
+Neither intelligibility nor grammatical correctness are not taken into account.
+
+For details, see the paper: http://www.aclweb.org/anthology/P02-1040.pdf
+"""
+
+_KWARGS_DESCRIPTION = """\
+Args:
+    name : str
+    batch_size : int, Batch size for openai completion.
+
+Optional Args:
+    None
+
+Functions:
+    _clean: clean special word in sentence.
+    _compute_single: compute bleu score for single prediction with its references
+
+Examples:
+    >>> from datasets import Dataset
+    >>> import rageval as rl
+    >>> sample = {
+    ...     "answers": [
+    ...         "Hello, world!",
```

Comment:
use more formal case as this is not only for test, but also an example for users.

Reply:
The cases by huggingface is
```
>>> predictions = [
...     ["hello", "there", "general", "kenobi"],
...     ["foo", "bar", "foobar"]
... ]
>>> references = [
...     [["hello", "there", "general", "kenobi"], ["hello", "there", "!"]],
...     [["foo", "bar", "foobar"]]
... ]
```
this case is not a whole sentence which is different from our input.
I will change it to cases from the original paper.